include/exadg/structure/preconditioners/multigrid_preconditioner.cpp

```diff
@@ -70,7 +70,7 @@ template<int dim, typename Number>
 void
 MultigridPreconditioner<dim, Number>::update()
 {
-  if(nonlinear)
+  if(nonlinear or data.unsteady)
```

Comment:
I have just seen that we do it exactly the same way in the momentum operator for incompressible flows. So should be fine as is. I think the responsibility is here in the hands of the user, who has to specify how often to update the preconditioner in the application file. If no update is needed (for certain model problems), the user has to deactivate this in the application to achieve optimal performance.

Reply:
This way, the steady linear solver does not do the update - as it would otherwise do the update twice (in a row, basically)!

Reply:
The cases where we do additional inits are now:
Steady nonlinear solver (+1 initially)
Unsteady nonlinear solver (+ 1 initially)
QuasiStatic(+1 initially)
One can also not skip the update in the first (quasi-)time step, since the update might be triggered every *n* Newton iterations or once the solver has converged.
One solution would be to add another flag to the `update` in the Newton solver, which checks if we are in the very first step after init! What do you think? ... changes some other files as well though ... then we could also think about checking the case `update_once_newton_converged` and the other flags (doing two head-to-head updates basically, with updated iterate/solution vector potentially?). Sounds more like a separate PR to me ... updating that Newton flags and (quasi-)time steppers?

Reply:
Currently, I think we should not let the Newton solver take the responsibility for this. I think this aspect is a convention between the spatial operator (initialization of solvers/preconditioners) and the time integrator (where we have the modulo operator, including a convention regarding the first time step for which we want to initiate an update).